src/hotspot/share/jvmci/jvmciRuntime.cpp

```diff
@@ -578,6 +581,8 @@ JRT_ENTRY(jlong, JVMCIRuntime::invoke_static_method_one_arg(JavaThread* current,
 JRT_END
 
 JRT_LEAF(void, JVMCIRuntime::log_object(JavaThread* thread, oopDesc* obj, bool as_string, bool newline))
+  assert(thread == JavaThread::current(), "pre-condition");
```

Comment:
`thread` seems unused in this function and so it is not obvious it has to be the current thread.